### PR TITLE
AWS security hub auth upgrade

### PR DESCRIPTION
**Pull request by infernuxmonster**
Required items, please complete
   
   Change(s):
   - Added new template for setting up IdP in AWS
   - Updated python-script to use Managed Identity of the function and use "AssumeRoleWithWebIdentity"
   - Updated ZIP-package with new script, packages
   - Updated readme.md with new guidance
   - Updated deployment-template to account for the changes

   Reason for Change(s):
   - Authentication with access keys is not secure or robust - this changes allows for best practice authentication without having to rotate or store keys.

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - Tested in multiple environments and works - fixed multiple bugs with authentication timing out etc.

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: DataConnectors/AWS-SecurityHubFindings/azuredeploy_awssecurityhub.json [hunk 1]
```diff
@@ -163,15 +161,18 @@
...
                         "AWSRegionName": "[parameters('AWSRegionName')]",
 						"LogAnalyticsCustomLogName":"[parameters('LogAnalyticsCustomLogName')]",
-						"Schedule":"0 */10 * * * *",
+						"Schedule":"*/10 * * * *",
 						"FreshEventTimeStamp":"10",
-						"SecurityHubFilters":"",
+						"SecurityHubFilters":"{'SeverityLabel': [{'Value': 'HIGH', 'Comparison': 'EQUALS'},{'Value': 'CRITICAL', 'Comparison': 'EQUALS'},],'RecordState': [{'Value': 'ACTIVE', 'Comparison': 'EQUALS'}]}",
 						"LAURI": "[variables('LogAnaltyicsUri')]",											 
-                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion.zip?raw=true"
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion.zip?raw=true?raw=true",
```
Comment (sreedharande) [inline DataConnectors/AWS-SecurityHubFindings/azuredeploy_awssecurityhub.json:172]: @infernuxmonster - Please delete duplicate ?raw=true.

**Comment by v-spadarthi** [comment]
@infernuxmonster : We wanted to check on the status of PR #5255 . PR is pending from more than 60 days. Let us know if any assistance is required for this PR. As Per our standard operating procedures if no response is received in the next 7 business days we will close this PR. Thank you for your cooperation.

**Comment by NikTripathi** [comment]
@v-spadarthi Please do the needful

**Review by infernuxmonster — COMMENTED** [review]
Fixed the error reported with double raw=true

File: DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion/__init__.py [hunk 2]
```diff
@@ -3,23 +3,31 @@
 import dateutil
 import logging
 import boto3
+import gzip
+import io
```
[the remark this answers is not in the document]
Comment (infernuxmonster) [reply DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion/__init__.py:7]: Fixed in update file.

**Review by infernuxmonster — COMMENTED** [review]
Update all errors reported by the bot, or commented on why it's not a problem.

File: DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion/__init__.py [hunk 3]
```diff
@@ -40,9 +48,20 @@
...
+        default_azure_credential = DefaultAzureCredential(exclude_shared_token_cache_credential=True)
+        credential_chain = ChainedTokenCredential(managed_identity, azure_cli, default_azure_credential)
+        token_meta = credential_chain.get_token(client_id)
+        token = token_meta.token
+    except azure.core.exceptions.ClientAuthenticationError as error:
+        logging.info ("Authenticating to Azure AD: %s" % error)
     
     sentinel = AzureSentinelConnector(logAnalyticsUri, sentinel_customer_id, sentinel_shared_key, sentinel_log_type, queue_size=10000, bulks_number=10)
-    securityHubSession = SecurityHubClient(aws_access_key_id, aws_secret_acces_key, aws_region_name)
+    securityHubSession = SecurityHubClient(aws_role_arn, aws_role_session_name, aws_region_name, token)
```
[the remark this answers is not in the document]
Comment (infernuxmonster) [reply DataConnectors/AWS-SecurityHubFindings/AzFunAWSSecurityHubIngestion/__init__.py:63]: No, token is defined before this happens.

**Comment by infernuxmonster** [comment]
@v-spadarthi - I've updated everything now, the errors reported by the bot should not be an actual issue.

**Comment by v-marimanda** [comment]
@infernuxmonster  checks is falling please take a look. thanks!!!

**Comment by infernuxmonster** [comment]
>
> @infernuxmonster checks is falling please take a look. thanks!!!

Not sure what you want me to do here @v-marimanda - I've commented on both checks.

* https://github.com/Azure/Azure-Sentinel/pull/5255#discussion_r968142686

assumed_role_object is set to the output of sts_client.assume_role_with_web_identity:
```python
assumed_role_object=sts_client.assume_role_with_web_identity(
                RoleArn=self.role_arn,
                RoleSessionName=self.role_session_name,
                WebIdentityToken=self.web_identity_token
                )
```
After that we call the variable:
```python
credentials=assumed_role_object['Credentials']
```

* https://github.com/Azure/Azure-Sentinel/pull/5255#discussion_r968143821

Here as well, we set token in the following code:
```python
try:
        managed_identity = ManagedIdentityCredential()
        azure_cli = AzureCliCredential()
        default_azure_credential = DefaultAzureCredential(exclude_shared_token_cache_credential=True)
        credential_chain = ChainedTokenCredential(managed_identity, azure_cli, default_azure_credential)
        token_meta = credential_chain.get_token(client_id)
        token = token_meta.token
    except azure.core.exceptions.ClientAuthenticationError as error:
        logging.info ("Authenticating to Azure AD: %s" % error)
```

Then we call it a few lines later:
```python
    securityHubSession = SecurityHubClient(aws_role_arn, aws_role_session_name, aws_region_name, token)
```

In both of these instances the variables are defined and set inside a try-catch (except), maybe that is the issue, but the idea is that the script should throw errors if it fails to set the token or the assumed_role_object because they are vital to the running of the script.

**Comment by v-marimanda** [comment]
@sreedharande can you please the review the pr. thanks!!!

**Comment by v-spadarthi** [comment]
@sreedharande can you please the review the pr. thanks!!!

**Comment by v-dvedak** [comment]
@sreedharande , Can you please review the PR?

**Comment by v-marimanda** [comment]
@sreedharande , Can you please review the PR?

**Comment by infernuxmonster** [comment]
Hey, sorry, missed this. Will fix ASAP.

tor. 8. sep. 2022, 22:05 skrev NikTripathi ***@***.***>:

**Comment by v-marimanda** [comment]
@sreedharande , Can you please review the PR?

**Comment by v-marimanda** [comment]
@sreedharande , Can you please review the PR?

**Comment by v-spadarthi** [comment]
@sreedharande , Can you please review the PR?

**Comment by v-spadarthi** [comment]
@sreedharande : Can you please review the PR?

**Pull request merged**